scenes/game/local_game.gd

```diff
@@ -1517,30 +1517,41 @@ class Player:
 	func discard(card_ids : Array):
 		var events = []
 		for discard_id in card_ids:
+			var found_card = false
+
 			# From hand
 			for i in range(len(hand)-1, -1, -1):
 				var card = hand[i]
 				if card.id == discard_id:
 					hand.remove_at(i)
 					events += add_to_discards(card)
 					on_hand_remove_public_card(discard_id)
+					found_card = true
 					break
+			if found_card: continue
```

Comment:
This is fine and saves time, but I wouldn't really expect it to take that much longer to look through like 20 things max.

Reply:
I thought there was a potential issue when I was writing my test cases, but I think it was just because I accidentally reused the card ID of something I had already put in gauge (so it went on to check the gauge card IDs but got a NPE because that card object didn't exist anymore)

Reply:
Yea, there's some weirdness in the tests around reusing the ids.